### PR TITLE
FISH-8286 Backport Date Fix

### PR DESCRIPTION
**Pull request by kalinchan**
See JIRA on testing instructions.

### CODE REVIEW

File: orbmain/src/main/java/com/sun/corba/ee/impl/io/IIOPInputStream.java [hunk 1]
```diff
@@ -1193,6 +1194,34 @@ private synchronized Object inputObject(Class clz, ClassInfoCache.ClassInfo cinf
...
+    // In JDK9, the Date class was modified to call defaultWriteObject(), which doesn't actually do anything but
+    // set the "defaultWriteObjectCalled" flag in the serialization of the class; unfortunately, the
+    // deserialization logic cannot handle that case and throws an exception. We are therefore employing this
+    // workaround to allow Date objects to be read across the wire between JDK8 and later JDKs.
+    // Fixing it will involve massive code simplification in order to eliminate duplicate code and allow for
+    // implementation of a simpler state machine that depends on the data sent rather than how it was produced.
+    private boolean readDefaultWriteObjectCalledFlag() throws IOException {
+        boolean sentDefaultWriteObjectCalled = readBoolean();
+
+        if (isDateClassWorkaroundRequired()) return getSimulatedDefaultWriteObjectCalledFlag();
```
Comment (breakponchito) [inline orbmain/src/main/java/com/sun/corba/ee/impl/io/IIOPInputStream.java]: Could you add the return statement on a new line? I mean that is better for legibility on the code
```suggestion
        if (isDateClassWorkaroundRequired()) 
            return getSimulatedDefaultWriteObjectCalledFlag();
```

**Review by breakponchito — APPROVED** [review]
LGTM

**Comment by breakponchito** [comment]
I'm ok with the approach and we need to create a new ticket to continue to improve implementation and use correct way to serialize and deserialize Date class

**Review by luiseufrasio — APPROVED** [review]
LGTM!

**Pull request merged**